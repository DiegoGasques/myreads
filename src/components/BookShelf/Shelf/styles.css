.shelf {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 250px;
  max-width: 98vw;
}
