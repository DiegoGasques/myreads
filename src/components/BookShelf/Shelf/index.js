import React, { Component } from "react";
import PropTypes from "prop-types";
import Carousel from "react-alice-carousel";

import "react-alice-carousel/lib/alice-carousel.css";
import "./styles.css";

class Shelf extends Component {
  responsive = {
    0: { items: 1 },
    600: { items: 2 },
    950: { items: 3 },
    1324: { items: 4 }
  };

  handleOnDragStart = e => e.preventDefault();

  render() {
    const { children } = this.props;
    return (
      <div className="shelf">
        <i class="fas fa-angle-left" />
        <Carousel
          mouseDragEnabled
          items={children(this.handleOnDragStart)}
          responsive={this.responsive}
        />
        <i class="fas fa-angle-right" />
      </div>
    );
  }
}

Shelf.propTypes = {
  children: PropTypes.node
};

export default Shelf;
