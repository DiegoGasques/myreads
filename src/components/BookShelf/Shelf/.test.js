import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Shelf from "./index";

it("Should match the component snapshot", () => {
  const wrapper = shallow(<Shelf>{() => []}</Shelf>);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
