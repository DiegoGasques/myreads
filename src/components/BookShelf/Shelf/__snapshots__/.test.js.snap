// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the component snapshot 1`] = `
<div
  className="shelf"
>
  <i
    className="fas fa-angle-left"
  />
  <AliceCarousel
    autoPlay={false}
    autoPlayDirection="ltr"
    autoPlayInterval={250}
    buttonsDisabled={false}
    disableAutoPlayOnAction={false}
    dotsDisabled={false}
    duration={250}
    fadeOutAnimation={false}
    infinite={true}
    items={Array []}
    keysControlDisabled={false}
    mouseDragEnabled={true}
    playButtonEnabled={false}
    preventEventOnTouchMove={false}
    responsive={
      Object {
        "0": Object {
          "items": 1,
        },
        "1324": Object {
          "items": 4,
        },
        "670": Object {
          "items": 2,
        },
        "980": Object {
          "items": 3,
        },
      }
    }
    showSlideInfo={false}
    slideToIndex={0}
    stagePadding={Object {}}
    startIndex={0}
    stopAutoPlayOnHover={true}
    swipeDisabled={false}
  />
  <i
    className="fas fa-angle-right"
  />
</div>
`;
